### PR TITLE
PROD-448 Debounce Second Order Selection Tracking

### PR DESCRIPTION
**Pull request by neel-ds**
- Description
It adds debounce to avoid too many requests at same time if user drags the slider continuously. It will delay the tracking call by 0.5 secs after percentage value changes.

- What are the steps to test that this code is working?
Answer second order and drag slider continuously. Verify the events recorded in mixpanel dashboard. It shouldn't hit too many requests at same time.

- Screen shots or recordings for UI changes

### CODE REVIEW

**Review by marvinmarnold — COMMENTED** [review]
Can you convince me that debounce is correct? It may still track values if the user is holding down on the mouse, but hasn't selected. We don't care about intermediary slider values until the user does click up.

**Comment by neel-ds** [comment]
> Can you convince me that debounce is correct? It may still track values if the user is holding down on the mouse, but hasn't selected. We don't care about intermediary slider values until the user does click up.

Answered in Linear

File: app/components/Deck/Deck.tsx [hunk 1]
```diff
@@ -219,6 +219,13 @@ export function Deck({
       }
 
       if (currentQuestionStep === QuestionStep.PickPercentage) {
+        trackQuestionAnswer(
```
Comment (marvinmarnold) [inline app/components/Deck/Deck.tsx:222]: Why does the tracking also need to happen here? I'm not clear when this gets invoked.

Comment (neel-ds) [reply app/components/Deck/Deck.tsx:222]: This will be invoked when a user clicks on Chomp button after selecting the second order.

Comment (marvinmarnold) [reply app/components/Deck/Deck.tsx:222]: Won't this lead to double tracking? It will have already been tracked once on click up.

Comment (neel-ds) [reply app/components/Deck/Deck.tsx:222]: It won't do double tracking as I removed the tracking when users drag the slider.

**Pull request merged**